### PR TITLE
WIP: Fail pod creation if resolv.conf too long

### PR DESCRIPTION
**Pull request by erictune**
needs e2e testing.

### CODE REVIEW

**Comment by erictune** [comment]
Surfaces to user the misconfiguration issue described in #9079

**Comment by bgrant0607** [comment]
cc @thockin

**Comment by thockin** [comment]
Idea LGTM

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -869,6 +877,19 @@ func (kl *Kubelet) GenerateRunContainerOptions(pod *api.Pod, container *api.Cont
 			return nil, err
 		}
 	}
+	// TODO(erictune): move into each container runtime, so we don't
+	// do this check on operating systems that don't use glibc (BSD, Windows).
+	// Also, this might be different with ipv6?
+	// TODO(erictune): get docker and rkt to warn about exceeding this limit?
+	if len(opts.DNSSearch) > glibcMaxResolvConfSearchEntries {
+		return nil, fmt.Errorf("Cluster DNS configuration error: more than %d search entries required", glibcMaxResolvConfSearchEntries)
```
Comment (ArtfulCoder) [inline pkg/kubelet/kubelet.go]: should the text use "specified" instead of "required"

File: pkg/kubelet/kubelet.go [hunk 2]
```diff
@@ -869,6 +877,19 @@ func (kl *Kubelet) GenerateRunContainerOptions(pod *api.Pod, container *api.Cont
...
+	}
+	if len(opts.DNS) > glibcMaxResolvConfNameserverEntries {
+		return nil, fmt.Errorf("Cluster DNS configuration error: more than %d nameserver entries required", glibcMaxResolvConfNameserverEntries)
+	}
```
Comment (ArtfulCoder) [inline pkg/kubelet/kubelet.go]: should the text use "specified" instead of "required"

File: pkg/kubelet/kubelet.go [hunk 3]
```diff
@@ -869,6 +877,19 @@ func (kl *Kubelet) GenerateRunContainerOptions(pod *api.Pod, container *api.Cont
...
+	if len(strings.Join(opts.DNSSearch, " ")) > glibcMaxResolvConfSearchChars {
+		return nil, fmt.Errorf("Cluster DNS configuration error: search string would exceed %d chars", glibcMaxResolvConfSearchChars)
+	}
```
Comment (ArtfulCoder) [inline pkg/kubelet/kubelet.go]: should the text say "exceeds" instead of "would exceed"

**Comment by ArtfulCoder** [comment]
helps with #8512

**Comment by erictune** [comment]
Fixed comments.
Working on test.

**Comment by erictune** [comment]
Not sure when I'll get back to writing a test for this.

**Pull request closed without merging**